### ISSUE TITLE
Unable to spot error in schema

### ISSUE THREAD

**Issue by sptrakesh** [tristanpenman/valijson#128]
Hello,

I am trying out this library for schema validation, and ran into an issue with one of my unit tests.  I found the following schema while searching for samples, and most online tools I tested it with, it seems to be valid.  However, valijson rejects it with `Expected numeric value for minimum constraint.` error.  Any idea where the error in the schema is?

```
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "description": "A product from Acme's catalog",
    "properties": {
      "id": {
        "description": "The unique identifier for a product",
        "type": "integer"
      },
      "name": {
        "description": "Name of the product",
        "type": "string"
      },
      "price": {
        "exclusiveMinimum": true,
        "minimum": 0,
        "type": "number"
      },
      "tags": {
        "items": {
          "type": "string"
        },
        "minItems": 1,
        "type": "array",
        "uniqueItems": true
      }
    },
    "required": ["id", "name", "price"],
    "title": "Product",
    "type": "object"
  }
```

**Comment by tristanpenman** [comment]
Thanks for raising this issue. There is definitely a bug in how Valijson handles exclusive minimum and exclusive maximum constraints. I'll try to fix this ASAP and update this issue once the changes have been pushed.

**Comment by tristanpenman** [comment]
I just had a closer look at this, and realised what is going. The schema that you're using looks like a draft 4 schema. In JSON Schema draft 4, the `exclusiveMinimum` property was a boolean value that changed how the `minimum` property was handled. So your schema would be valid if the SchemaParser was initialised in `kDraft4` mode. 

In later JSON Schema drafts, `exclusiveMinimum` was changed to be a numeric property. The same applies for maximums. This is all documented towards the end of this page in JSON Schema docs: https://json-schema.org/understanding-json-schema/reference/numeric.html

That said, the error message is incorrect, and should refer to `exclusiveMinimum` instead of `minimum`. That's just a poor copy+paste mistake on my part!

This has been fixed in 4897d10.

Feel free to re-open this ticket if this doesn't resolve the issue.
